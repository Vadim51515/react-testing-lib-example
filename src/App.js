// import logo from './logo.svg';
// import './App.css';
// import {
//     useEffect,
//     useState,
// } from 'react';
//
// function App() {
//     const [data, setData] = useState(null);
//     const [toggle, setToggle] = useState(false);
//     const [value, setValue] = useState('');
//
//     useEffect(() => {
//         setTimeout(() => {
//             setData({ name: 'Вадим' });
//         }, 100);
//     }, []);
//
//     const onClick = () => setToggle(prev => !prev);
//
//     return (
//         <div className='App'>
//             <h1>Hello World</h1>
//             <button  data-testid='toggle-btn' onClick={onClick}>Нажми на меня</button>
//             <input
//                 onChange={e => setValue(e.target.value)}
//                 type='text'
//                 data-testid='main-input'
//                 placeholder='в меня можно ввести текст'
//             />
//             {data?.name && <p>{data?.name}</p>}
//             {toggle && <p data-testid='toggle-elem'>Тогл сейчас в положении вкл</p>}
//             <p data-testid='value-input'>{value}</p>
//         </div>
//     );
// }
//
// export default App;

import React from 'react';
import AppRouter from './router/AppRouter';
import Navbar from './components/Navbar/Navbar';

const App = () => {
    return (
        <div>
            <Navbar />
            <AppRouter />
        </div>
    );
};

export default App;
