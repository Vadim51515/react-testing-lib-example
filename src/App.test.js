import {
    act,
    fireEvent,
    render,
    screen,
} from '@testing-library/react';
import App from './App';
import userEvent from '@testing-library/user-event';

describe('Тестиреум App', () => {

    test('Провереям основные элементы на странице', () => {
        render(<App />);
        const titleText = screen.getByText(/hello world/i);
        const btn = screen.getByRole('button');
        const input = screen.getByPlaceholderText(/в меня можно ввести текст/i);

        expect(titleText).toBeInTheDocument();
        expect(btn).toBeInTheDocument();
        expect(input).toMatchSnapshot();
    });

    test('Проверяем, что нет ничего лишнего', () => {
        render(<App />);
        const titleText = screen.queryByText(/hello2/i);
        expect(titleText).toBeNull();
    });

    test('Проверяем, асинхронный вывод имени', async () => {
        render(<App />);
        const nameText = await screen.findByText(/вадим/i);
        expect(nameText).toBeInTheDocument();
    });

    test('Клик по тоглу', () => {
        render(<App />);
        const btn = screen.getByTestId('toggle-btn');

        // Элемента ЕЩЕ нет на странице
        expect(screen.queryByTestId('toggle-elem')).toBeNull();

        fireEvent.click(btn);
        // Элемент с нужным тестом есть на странице
        expect(screen.queryByTestId('toggle-elem')).toHaveTextContent(/Тогл сейчас в положении вкл/i);

        fireEvent.click(btn);
        // Элемента УЖЕ нет на странице
        expect(screen.queryByTestId('toggle-elem')).toBeNull();
    });

    test('Вввод значения в инпут', () => {
        render(<App />);

        const input = screen.queryByTestId('main-input');
        expect(screen.queryByTestId('value-input')).toContainHTML('');

        const newValue = '123123';
        fireEvent.input(input, {
            target: { value: newValue },
        });
        expect(screen.queryByTestId('value-input')).toContainHTML(newValue);
    });

    test('Вввод значения в инпут с помощью userEvent',  () => {
        render(<App />);

        const input = screen.queryByTestId('main-input');
        expect(screen.queryByTestId('value-input')).toContainHTML('');

        const newValue = '123123';
        act(() => {
             userEvent.type(input, newValue)
        })
        expect(screen.queryByTestId('value-input')).toContainHTML(newValue);
    });

});

