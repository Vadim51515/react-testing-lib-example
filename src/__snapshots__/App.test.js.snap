// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестиреум App Провереям основные элементы на странице 1`] = `
<input
  data-testid="main-input"
  placeholder="в меня можно ввести текст"
  type="text"
/>
`;
